libmamba/src/core/package_info.cpp

```diff
@@ -273,6 +273,11 @@ namespace mamba
     {
     }
 
+    bool PackageInfo::operator==(PackageInfo const& other) const
+    {
+        return sha256 == other.sha256;
```

Comment:
I think it might be a  low probability but there could be a case where multiple packageInfos have the same sha256. In my PR  I checked the equality of each of the fields - https://github.com/mamba-org/mamba/pull/1847/files#diff-c93060e224d433cee7ee3ad6f251e0c1577515984b880e21665e16ac5b45f3c2R72
Also, would it be possible that sha256 is empty for some of the packages and only md5 is set?

Reply:
>  think it might be a low probability but there could be a case where multiple packageInfos have the same sha256.

It's considered small enough to be safe, otherwise that would be a big security problem!

>  Also, would it be possible that sha256 is empty for some of the packages and only md5 is set?

We could use both.

> In my PR I checked the equality of each of the fields

I've also written code that does the `hash` for `std::tuple` so that we can use `std::tie`, but then i saw there was a `std::set` and many complicated things so I went with this.